### PR TITLE
Rust 1.76 toolchain

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

File: tools/make/ffi.toml [hunk 1]
```diff
@@ -68,7 +68,7 @@ description = "Build ICU4X CAPI for Cortex"
 category = "ICU4X FFI"
 dependencies = ["install-nightly", "install-cortex-8"]
 toolchain = "${ICU4X_NIGHTLY_TOOLCHAIN}"
-env = { RUSTFLAGS = "-Ctarget-cpu=cortex-m33 -Cpanic=abort -Copt-level=s" }
```
Comment (Manishearth) [inline tools/make/ffi.toml:71]: question: why remove this?

File: provider/datagen/src/tests/data.rs [hunk 2]
```diff
@@ -21,7 +21,7 @@ impl DatagenProvider {
             .get_or_init(|| Self {
                 source: SourceData {
                     cldr_paths: Some(Arc::new(CldrCache::from_serde_cache(SerdeCache::new(AbstractFs::Memory(
```
Comment (sffc) [inline provider/datagen/src/tests/data.rs:23]: Why is there a diff in this file?

File: tools/make/ffi.toml (hunk 1 above)
Comment (robertbastian) [reply tools/make/ffi.toml:71]: I got an error locally saying that the cpu is unknown

File: provider/datagen/src/tests/data.rs (hunk 2 above)
Comment (robertbastian) [reply provider/datagen/src/tests/data.rs]: there seems to be some instability in the gen script, which I ran locally. I'll fix in a separate PR

**Review by sffc — DISMISSED** [review]
lgtm for me but would like @Manishearth to lgtm the `cortex-m33` change

**Review by Manishearth — APPROVED** [review]

**Pull request merged**